### ISSUE TITLE
`E0061` was not detected

### ISSUE THREAD

**Issue by Stzx** [intellij-rust/intellij-rust#5681]
## Environment

* **IntelliJ Rust plugin version:** `0.2.126.SNAPSHOT-201-dev`
* **Rust toolchain version:** `rustc 1.46.0-nightly (feb3536eb 2020-06-09)` 
* **IDE name and version:** `IntelliJ IDEA 2020.1.2 (Ultimate Edition)`
* **Operating system:** `Windows 10 10.0`

## Problem description
```rust
use rand::{Rng, thread_rng};

fn main() {
    let mut rng = thread_rng();

    println!("{}", rng.gen_bool()); // No error here
}
```

```rust
error[E0061]: this function takes 1 argument but 0 arguments were supplied
 --> src\main.rs:6:24
  |
6 |     println!("{}", rng.gen_bool());
  |                        ^^^^^^^^- supplied 0 arguments
  |                        |
  |                        expected 1 argument
```

**Comment by Stzx** [comment]
Related to this line of code:

https://github.com/intellij-rust/intellij-rust/blob/4c55740646481883dd032b778aa15dad966f3523/src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt#L1171

**Comment by Kobzol** [comment]
E0061 is turned off for trait methods because of errors in name resolution. It's stated in the original commit/PR that introduced the a
E0061 annotation. I checked the example that was failing a few years ago and it still fails, so it should probably stay turned off for now.

**Comment by vlad20012** [comment]
Oh, [`test ignores trait implementations E0061` test](https://github.com/intellij-rust/intellij-rust/blob/ed366d911bfecca3ed9cd875b06301c8314fa1b6/src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt#L336) is really interesting! I think I still don't understand the Rust type inference algorithms because I don't understand how it works!

**Comment by vlad20012** [comment]
Oh, I found the reason. We don't add implicit `T: Sized` bounds to generics.
